File: Data Structures & Algorithms/is-anagram/submission-3.java
class Solution {
     int[] count = new int[26];
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
       
       for(char ch:s.toCharArray()){
        count[ch-'a']++;
       }
       for(char ch:t.toCharArray()){
        count[ch-'a']--;
       }

       for(int num: count){
        if(num!=0){
            return false;
        }
       }
       return true;

    }
}

File: Data Structures & Algorithms/boats-to-save-people/submission-3.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        n=len(people)-1
        left,right=0,n
        count=0
        while left<=right:
            if people[left]+people[right]==limit:
                left+=1
                right-=1
            elif people[left]+people[right]>limit:
                right-=1
            else:
                left+=1
                right-=1
            count+=1
        return count


        